app/controllers/transcriptions_controller.rb

```diff
@@ -49,6 +49,9 @@ def export
     data_storage.zip_transcription_files(@transcription) do |zip_file|
       send_export_file zip_file
     end
+
+  rescue DataExports::NoStoredFilesFoundError => e
```

Comment:
Why does this need to be rescued here and not just propagated up to be caught in the ErrorExtender?

Reply:
If it's caught by `ErrorExtender`, it's handled as an internal server error (500), and the error details are not included.

Reply:
If you add it as a `rescue_from` entry, it can render whatever you want. It should only slip all the way to StandardError and an internal server error if nothing else catches it.

Reply:
So the weird thing is that when I add `DataExports::NoStoredFilesFoundError` as a `rescue_from` entry (either in the `TranscriptionsController` or in the `ErrorExtender`), it throws an error:
```
NameError:
  uninitialized constant DataExports::NoStoredFilesFoundError
```

Adding a `rescue_from` for the `NoExportableTranscriptionsError` works fine though. Any idea why?

Reply:
Not sure why it won't load it from the module, but you can just move the error to be defined in the DataStorage class and call the error like `raise DataExports::DataStorage::NoStoredFilesFoundError`

Reply:
Huh, yeah that doesn't work either. When i put the `rescue_from` in either the ErrorExtender or the TranscriptionsController, I get a similar error:
```
NoMethodError:
  undefined method `NoStoredFilesFoundError' for DataExports::DataStorage:Class
```
If you think it's worth it, I can do more investigation on why an error thrown within a service class isn't picked up within the controller, but it _does_ work to just rescue the error within the export method.

Reply:
Seems to work here. data_storage.rb:
```
module DataExports
  class DataStorage
    class NoStoredFilesFoundError < StandardError; end
    ...
```
error_extender.rb:
```
    ... (other rescue_froms)
    rescue_from DataExports::DataStorage::NoStoredFilesFoundError, with: :render_jsonapi_not_found
```
data_storage_spec.rb:
```
      expect { data_storage.zip_transcription_files(transcription) }.to raise_error(DataExports::DataStorage::NoStoredFilesFoundError)
```

Reply:
Oh gosh wow, I just made a dumb mistake and overlooked the fact that I hadn't included a comma after the error type in the `rescue_from` ... 😓I guess I was not expecting it to work, and so I didnt dig into the error just assumed it was another iteration of the error I got before. Thanks for checking, you're fantastic. Still, funny that it doesn't work when the error type is declared outside of a class (because I did go back and try that again just now, and I definitely got the syntax right). Gonna take note of that for future reference.